prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -296,11 +296,21 @@ public APIResult update(HttpServletRequest request, String type, String entityNa
     protected APIResult update(InputStream inputStream, String type, String entityName,
                                String colo, Boolean skipDryRun) {
         checkColo(colo);
+        try {
+            EntityType entityType = EntityType.getEnum(type);
+            Entity entity = deserializeEntity(inputStream, entityType);
+            return update(entity, type, entityName, skipDryRun);
+        } catch (IOException | FalconException e) {
+            LOG.error("Update failed", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
```

Comment:
HTTP 400 Bad Request code is returned when the request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications. I think exceptions from deserializeEntity should return BAD_REQUEST and any exceptions because of update should return INTERNAL_SERVER_ERROR.